### PR TITLE
default_write_temperature option

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: Currently SST files that aren't applicable to last_level_temperature nor file_temperature_age_thresholds are written with temperature kUnknown, which is a little weird and doesn't support CF-based tiering. The default_temperature option only affects how kUnknown is interpreted for stats.

This change adds a new per-CF option default_write_temperature that determines the temperature of new SST files when those other options do not apply.

Also made a change to ignore last_level_temperature with FIFO compaction, because I found that could lead to an infinite loop in compaction.

Needed follow-up: Fix temperature handling with external file ingestion

Test Plan: unit tests extended appropriately. (Ignore whitespace changes when reviewing.)

### CODE REVIEW

File: db/external_sst_file_ingestion_job.cc [hunk 1]
```diff
@@ -160,9 +160,9 @@ Status ExternalSstFileIngestionJob::Prepare(
       TEST_SYNC_POINT_CALLBACK("ExternalSstFileIngestionJob::Prepare:CopyFile",
                                nullptr);
       // CopyFile also sync the new file.
-      status =
-          CopyFile(fs_.get(), path_outside_db, path_inside_db, 0,
-                   db_options_.use_fsync, io_tracer_, Temperature::kUnknown);
+      status = CopyFile(fs_.get(), path_outside_db, path_inside_db, 0,
```
Comment (jowlyzhang) [inline db/external_sst_file_ingestion_job.cc:163]: The external file ingestion job supports a per ingestion temperature option: https://github.com/facebook/rocksdb/blob/a43481b3d06f132e0b4d0d9d272d59eedbf09065/include/rocksdb/db.h#L134

This option seems to be missed from this `CopyFile` invocation but was used in some other places like creating `FileMetaData`. Shall we consolidate the priority of these two options: per ingestion file temperature, per column family default write file temperature. And use the chosen one consistently in all places?

File: include/rocksdb/advanced_options.h [hunk 2]
```diff
@@ -818,12 +818,20 @@ struct AdvancedColumnFamilyOptions {
   Temperature bottommost_temperature = Temperature::kUnknown;
   Temperature last_level_temperature = Temperature::kUnknown;
 
+  // EXPERIMENTAL
+  // When no other option such as last_level_temperature determines the
+  // temperature of a new SST file, it will be written with this temperature,
+  // which can vary by ColumnFamily.
```
Comment (jowlyzhang) [inline include/rocksdb/advanced_options.h:824]: very nit: "ColumnFamily/column family"

**Review by jowlyzhang — APPROVED** [review]
LGTM, thank you for making this change! Would you mind adding a change log for this?

File: db/external_sst_file_ingestion_job.cc (hunk 1 above)
Comment (pdillinger) [reply db/external_sst_file_ingestion_job.cc:163]: There's quite a bit to unravel and fix with temperatures and ingestion, so I'm going to defer that to a follow-up PR, and put a hefty FIXME here in this PR.

**Pull request closed without merging**